pkg/cloud/services/ec2/instances.go

```diff
@@ -738,6 +745,12 @@ func (s *Service) DetachSecurityGroupsFromNetworkInterface(groups []string, inte
 	return nil
 }
 
+func (s *Service) shouldUseEksAMI() bool {
+	gvk := s.scope.InfraCluster().GetObjectKind().GroupVersionKind()
+
+	return gvk.Kind == "AWSManagedCluster"
```

Comment:
Should this just be `EKSCluster` instead of `AWSManagedCluster`?

Reply:
https://github.com/kubernetes-sigs/cluster-api-provider-aws/pull/1724 introduces `AWSManagedCluster` and `AWSManagedControlPlane`. Though after deciding to put most of the logic in the controlplane provider, this should be `AWSManagedControlPlane` actually.